api/controllers/user.js

```diff
@@ -0,0 +1,64 @@
+import bcrypt from 'bcryptjs';
+import models from '../../database/models';
+import { validationResult } from 'express-validator';
+
+const user = {
+    // To aid testing 
+    createUser: (req, res) => {
+        const { name, email, password } = req.body
+        const errors = validationResult(req);
+
+        if (!errors.isEmpty()) {
+          return res.status(422).json({ errors: errors.array() });
+        }
+
+        return models.sequelize.sync({force: false}).then(() => {
+            const saltRound = 10;
+            bcrypt.genSalt(saltRound, (err, salt) => {
+                bcrypt.hash(password, salt, (err, hash) => {
+                    models.user.create({name, email, password: hash}).then((user) => {
+                        const payload = {
+                            email: user.email,
+                            password: user.password
+                        }
+                        res.status(200).json({data: payload})
+                    }).catch((error) => {
+                        if (error) {
+                            res.status(409).json({message: 'User already exist'});
+                        }
+                    });
+                });
+            });
+        });
+    },
+
+    adminLogin: (req, res) => {
+        const { email, password } = req.body
+        const errors = validationResult(req);
+
+        if (!errors.isEmpty()) {
+          return res.status(422).json({ errors: errors.array() });
+        }
+
+        models.user.findOne({where: { email }}).then((user) => {
+            if (user && user.dataValues.email === email && user.dataValues.isAdmin === true) {
```

Comment:
I think the test for email here is a bit redundant. I suggest we use:

```suggestion
            if (user &&  user.dataValues.isAdmin === true) {
            
```

Reply:
Okay, noted, thanks.